opentracing/src/main/java/io/opentracing/Span.java

```diff
@@ -52,6 +57,14 @@
    * Returns null if no entry found, or baggage is not supported in the current implementation.
    */
   String getBaggageItem(String key);
+  
+  /**
+   * Useful for acquiring the set of baggage keys so that clients of the Span have the ability
+   * to access the corresponding baggage value.
+   * 
+   * @return {@link Set} of baggage keys
+   */
+  Set<String> getBaggageKeys();
```

Comment:
In Tracer.inject, I want to enumerate through the existing key/value pairs and push them into the corresponding carrier without access internal state of the Span implementation.  Want to avoid this:

span.baggage.entrySet()

for (String key : span.getBaggageKeys()) {
  carrier.put(key, span.getBaggageItem(value));
}

I want to keep inject code as generic as possible and not code against a span implementation.